components/script/dom/gpubuffer.rs

```diff
@@ -202,3 +269,40 @@ impl GPUBufferMethods for GPUBuffer {
         *self.label.borrow_mut() = value;
     }
 }
+
+impl AsyncWGPUListener for GPUBuffer {
+    #[allow(unsafe_code)]
+    fn handle_response(&self, response: WebGPUResponse, promise: &Rc<Promise>) {
+        match response {
+            WebGPUResponse::BufferMapAsync(bytes) => {
+                if unsafe {
```

Comment:
The return type for `ArrayBuffer::create()` is `Result<(),()>` so there is no actual error to log here. Logging a custom message could still be helpful so I'll do that.

Reply:
ah ok. In cases like this `match` is still better (it's better in most cases), since you can match on `Err(())` to explicitly show that the error is `()` and avoid people asking weird questions :)